Sources/Views/MessageLabel.swift

```diff
@@ -301,10 +319,10 @@ open class MessageLabel: UILabel {
 
     private func parse(text: NSAttributedString) -> [NSTextCheckingResult] {
         guard enabledDetectors.isEmpty == false else { return [] }
-        let checkingTypes = enabledDetectors.reduce(0) { $0 | $1.textCheckingType.rawValue }
-        let detector = try? NSDataDetector(types: checkingTypes)
         let range = NSRange(location: 0, length: text.length)
-        let matches = detector?.matches(in: text.string, options: [], range: range) ?? []
+        let matches = enabledDetectors
+            .compactMap { parse($0, for: text, in: range) }
```

Comment:
This is worrisome to me 🤔 The old detector code took in all the checking types up front. We're now iterating and parsing the text each time. I'm not sure how `NSDataDetector` operates internally, but if this has gone from parsing 1 time to parsing N times -- we can't handle that overhead

Reply:
@SD10 May be I can refactor that part to put all `NSTextCheckingResult.CheckingType` and call `NSDataDetector` once. Like before

`NSDataDetector` is a subclass of `NSRegularExpression`